### ISSUE TITLE
com.twitter.heron.streamlet.impl.sources.ComplexSource cannot be cast to org.apache.heron.api.spout.ISpout

### ISSUE THREAD

**Issue by Vxider** [apache/incubator-heron#3242]
I have encountered an exception when using implementing 'custom source' followed by "SimplePulsarSourceTopology" in example. Here is the exception log:

```
java.lang.ClassCastException: com.twitter.heron.streamlet.impl.sources.ComplexSource cannot be cast to org.apache.heron.api.spout.ISpout
        at org.apache.heron.instance.spout.SpoutInstance.<init>(SpoutInstance.java:121)
        at org.apache.heron.instance.Slave.handleNewAssignment(Slave.java:154)
        at org.apache.heron.instance.Slave.handleNewPhysicalPlan(Slave.java:364)
        at org.apache.heron.instance.Slave.access$300(Slave.java:55)
        at org.apache.heron.instance.Slave$1.run(Slave.java:124)
        at org.apache.heron.common.basics.WakeableLooper.executeTasksOnWakeup(WakeableLooper.java:191)
        at org.apache.heron.common.basics.WakeableLooper.runOnce(WakeableLooper.java:110)
        at org.apache.heron.common.basics.WakeableLooper.loop(WakeableLooper.java:100)
        at org.apache.heron.instance.Slave.run(Slave.java:186)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
```

Is this error caused by the difference between the Heron application(org.apache.xxx) and the maven API(com.twitter.xxx) package name?

**Comment by nwangtw** [comment]
Looks possible. The package name was com.twitter.* before it was changed to org.apache.* since version 0.20.

My guess is that you are building old example against the new library.

**Comment by Vxider** [comment]
Thank you for your reply, the problem was solved when I downgraded the heron version to 0.17.5.
